app/export.go

```diff
@@ -62,6 +65,62 @@ var exportablePreferences = map[ComparablePreference]string{{
 }: "EmailInterval",
 }
 
+func (a *App) GetID() string {
+	return a.ID
+}
+func (a *App) SetID(id string) {
+	a.ID = id
+}
+
+func exportLogFile(attachment AttachmentImportData, exportFilename string) (*LogFormat, *model.AppError) {
+	createdAtTime, err := strconv.ParseInt(strconv.FormatInt(attachment.CreateAt, 10), 10, 64)
+	if err != nil {
+		return nil, model.NewAppError("parseTime", "Failed to parse createdAt time", nil, err.Error(), http.StatusInternalServerError)
+	}
+	updatedAtTime, err := strconv.ParseInt(strconv.FormatInt(attachment.UpdateAt, 10), 10, 64)
+	if err != nil {
+		return nil, model.NewAppError("parseTime", "Failed to parse updatedAt time", nil, err.Error(), http.StatusInternalServerError)
+	}
+	data := LogFormat{
+		Id:            attachment.Id,
+		CreatorId:     attachment.CreatorId,
+		PostId:        attachment.PostId,
+		ChannelId:     attachment.ChannelId,
+		CreateAt:      time.UnixMilli(createdAtTime),
+		UpdateAt:      time.UnixMilli(updatedAtTime),
+		Path:          *attachment.Path,
+		ThumbnailPath: attachment.ThumbnailPath,
+		PreviewPath:   attachment.PreviewPath,
+		Name:          attachment.Name,
+		Extension:     attachment.Extension,
+		Size:          attachment.Size,
+		MimeType:      attachment.MimeType,
+		Width:         attachment.Width,
+		Height:        attachment.Height,
+	}
+	return &data, nil
+}
+
+func (a *App) ExportAttachments(attachment AttachmentImportData, exportFilename string, missingAttachments *[]LogFormat, outPath string, zipWr *zip.Writer) *model.AppError {
+	isExist, err := a.FileExists(*attachment.Path)
+	if err != nil {
+		return err
+	}
+	if !isExist {
+		data, err := exportLogFile(attachment, exportFilename)
+		if err != nil {
+			return err
+		}
+		*missingAttachments = append(*missingAttachments, *data)
```

Comment:
Here you have created a pointer of missingAttachments but below you are doing this . Make this consistent

Reply:
Okk